### PR TITLE
refactor(field): migrate reference, array, image, file diff components to @sanity/ui

### PR DESCRIPTION
**Pull request by vicbergquist**
**Note:** builds on #2587, should be reviewed and merged after that one

### Description

Fixes some layout issues in the diff components, otherwise there are no visual changes, but the components are now using @sanity/ui. I would have separated this PR into two, but the reference and array diff components also required some changes in the layout components to work as expected.

### What to review

Test documents that uses these diff components in the review changes panel. Remove, add, and change content to test the refactored components. Document types and their paths:

- Arrays test (`Array of predefined options`): `/test/desk/arraysTest`
- Images test: `/test/desk/imagesTest`
- Files test: `/test/desk/filesTest`
- Reference test: `/test/desk/referenceTest`
- as well as any documents that contain any of these types

### Notes for release

- Migrated reference, array, image, and file diff components to @sanity/ui

### CODE REVIEW

File: packages/@sanity/field/src/types/image/diff/ImagePreview.tsx [hunk 1]
```diff
@@ -43,17 +103,20 @@ export function ImagePreview(props: ImagePreviewProps): React.ReactElement {
     .fit('max')
 
   const assetChanged = diff.fromValue?.asset?._ref !== diff.toValue?.asset?._ref
-  const imageWrapperClassName =
-    is === 'from' && assetChanged ? styles.imageWrapperChanged : styles.imageWrapper
+
   const metaAction = action === 'changed' ? undefined : action
 
   return (
```
Comment (rexxars) [inline packages/@sanity/field/src/types/image/diff/ImagePreview.tsx:109]: The image icons seems to be a bit cramped compared to what it used to be:

image
vs
image

The right hand side looks slightly _better_ in my opinion, but the left hand side are a bit too close to the icon?

**Review by rexxars — CHANGES_REQUESTED** [review]
Overall this looks good - I commented on a few optimizations where invoking React hooks is unnecessary, and there's a few visual regressions to address

File: packages/@sanity/field/src/types/image/diff/ImagePreview.tsx [hunk 2]
```diff
@@ -21,11 +22,70 @@ const imageBuilder = imageUrlBuilder(versionedClient)
...
+    </Flex>
+  </Card>
 )
 
+const ImageWrapper = styled.div`
+  height: 100%;
+  max-height: 190px;
+  position: relative;
+
+  /* Ideally the checkerboard component currently in the form builder should be made available and used here */
```
[the remark this answers is not in the document]
Comment (vicbergquist) [reply packages/@sanity/field/src/types/image/diff/ImagePreview.tsx:39]: Wrote a comment about it here: https://app.clubhouse.io/sanity-io/story/1693/ui-checkerboard-component

File: packages/@sanity/field/src/types/file/diff/FileFieldDiff.tsx [hunk 3]
```diff
@@ -86,10 +111,10 @@ export const FileFieldDiff: DiffComponent<ObjectDiff<File>> = ({diff, schemaType
       )}
 
       {nestedFields.length > 0 && (
-        <div className={styles.nestedFields}>
+        <Box marginTop={3}>
           <ChangeList diff={diff} schemaType={schemaType} fields={nestedFields} />
-        </div>
+        </Box>
       )}
```
[the remark this answers is not in the document]
Comment (vicbergquist) [reply packages/@sanity/field/src/types/file/diff/FileFieldDiff.tsx:117]: Ah didn't realise it happened here as well. I think I fixed this in the next PR, so will add the fix here as well. Thanks!

File: packages/@sanity/field/src/types/image/diff/ImagePreview.tsx (hunk 1 above)
Comment (vicbergquist) [reply packages/@sanity/field/src/types/image/diff/ImagePreview.tsx:109]: Looks like the container doesn't stretch properly, might be an issue fixed in the next PR, will look into it and include it here if that's the case

File: packages/@sanity/field/src/types/file/diff/FileFieldDiff.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (vicbergquist) [reply packages/@sanity/field/src/types/file/diff/FileFieldDiff.tsx:117]: not able to find this, can you share a link where it's happening?

File: packages/@sanity/field/src/types/image/diff/ImagePreview.tsx (hunk 1 above)
Comment (vicbergquist) [reply packages/@sanity/field/src/types/image/diff/ImagePreview.tsx:109]: was a padding issue combined with a hard px width for the arrow column. Fixing 😊

File: packages/@sanity/field/src/types/file/diff/FileFieldDiff.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (rexxars) [reply packages/@sanity/field/src/types/file/diff/FileFieldDiff.tsx:117]: Somehow I'm unable to reproduce :/ 

I did discover a bug while I was at it:
1. Add a file 
2. Add a description
3. Publish
4. Change the description
5. Review changes shows the asset changed, where it should only be showing the difference in description:
image

[the remark this answers is not in the document]
Comment (vicbergquist) [reply packages/@sanity/field/src/types/file/diff/FileFieldDiff.tsx:117]: Looks like the annotation tooltip is also buggy

[the remark this answers is not in the document]
Comment (vicbergquist) [reply packages/@sanity/field/src/types/file/diff/FileFieldDiff.tsx]: Found the problem. Looks like the bug was also in next, and caused by that `didAssetChange` wasn't being used as it should (and why it was unused). Fixing

[the remark this answers is not in the document]
Comment (vicbergquist) [reply packages/@sanity/field/src/types/file/diff/FileFieldDiff.tsx]: Fixed

**Review by rexxars — APPROVED** [review]
👍 LGTM

**Pull request merged**